### PR TITLE
Rename client.Tx to client.Txn.

### LINKED ISSUE #997: convert codebase from using client.KV to client.DB
This work is going to take place as a series of PRs which replace usage of `client.KV` with `client.DB`. Temporarily, the `client.DB` object will be created close to its usage using `client.KV.NewDB()`. Once the leaf modules are changed I'll work on pushing `client.DB` higher in the stack so that callers pass `client.DB` instead of `client.KV` to various modules. For example, `NewIDAllocator` currently takes a `client.KV` parameter but uses a `client.DB` internally.

As a final step, when `client.{KV,Txn}` is no longer being used outside of the `client` package itself I'll unexport those types and work on integrating their functionality directly into `client.{DB,Tx}`.

### PR DESCRIPTION
**Pull request by petermattis**
Mostly done using gorename with a few hand edits to comments.

Simplify txnSender slightly by making it a type alias for Txn.

Fixes #997.

### CODE REVIEW

**Comment by tamird** [comment]
LGTM

File: client/txn.go [hunk 1]
```diff
@@ -65,56 +63,52 @@ func (ts *txnSender) Send(ctx context.Context, call Call) {
...
 	wrapped      Sender
 	txn          proto.Transaction
-	txnSender    txnSender
 	haveTxnWrite bool // True if there were transactional writes
 	haveEndTxn   bool // True if there was an explicit EndTransaction
 }
 
-func newTx(db *DB, depth int) *Tx {
-	tx := &Tx{
+func newTxn(db *DB, depth int) *Txn {
```
Comment (tamird) [inline client/txn.go]: while you're here, make this take `db` by value?

Comment (petermattis) [inline client/txn.go]: Sure. Done.

File: client/db.go [hunk 2]
```diff
@@ -357,15 +357,15 @@ func (db *DB) Run(b *Batch) error {
...
 // transaction is automatically aborted if retryable returns any error aside
 // from recoverable internal errors, and is automatically committed
 // otherwise. The retryable function should have no side effects which could
 // cause problems in the event it must be run more than once.
 //
 // TODO(pmattis): Allow transaction options to be specified.
-func (db *DB) Tx(retryable func(tx *Tx) error) error {
-	return newTx(db, 1).exec(retryable)
+func (db *DB) Txn(retryable func(txn *Txn) error) error {
+	return newTxn(*db, 1).exec(retryable)
```
Comment (tschottdorf) [inline client/db.go]: I like putting an inline comment `/* depth */` in invocations that have an unguessable parameter.

Comment (petermattis) [inline client/db.go]: Unguessable? Or difficult to guess?

Done.

**Comment by tschottdorf** [comment]
LGTM

**Pull request merged**